src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts

```diff
@@ -224,10 +231,19 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 	public toggleEditMode(editMode?: boolean): void {
 		this.isEditMode = editMode !== undefined ? editMode : !this.isEditMode;
 		this.cellModel.isEditMode = this.isEditMode;
+		if (this.isEditMode === false) {
+			this.togglePreview(true);
+		}
 		this.updatePreview();
 		this._changeRef.detectChanges();
 	}
 
+	public togglePreview(showPreview?: boolean): void {
```

Comment:
toggle usually means you're switching it from its current value to the other boolean value which isn't really what's happening here. 

Instead I'd make doShowPreview into a property and then have the setter contain this logic.

Reply:
I agree that this should be modified. I will need your help.